src/hotspot/share/services/threadService.cpp

```diff
@@ -874,7 +874,10 @@ void ThreadSnapshot::initialize(ThreadsList * t_list, JavaThread* thread) {
   _sleep_ticks = stat->sleep_ticks();
   _sleep_count = stat->sleep_count();
 
-  _thread_status = java_lang_Thread::get_thread_status(threadObj);
+  // If thread is still attaching then threadObj will be NULL.
+  _thread_status = threadObj == NULL ? JavaThreadStatus::NEW
+                                     : java_lang_Thread::get_thread_status(threadObj);
```

Comment:
I like the use of `JavaThreadStatus::NEW` here.

Since L867 creates the _threadObj OopHandle, do you want to change both
uses of `threadObj` to `_threadObj()`? Is that still how we fetch the oop from
an OopHandle? Or even use `threadObj()`... Then we don't have to reason
about whether the `threadObj` oop is still good...

Reply:
I was thinking about the same. If we always use _threadObj() then there is no need to make sure the threadObj is still valid.

Reply:
I'm not quite sure what point is being made here. We don't have to do anything to "make sure the threadObj is still valid" as we don't have any code here that could expose a "naked oop". We only need/want to use a handle when we have to ensure the oop is preserved across calls. That is not the case here.

Reply:
I agree, it is not the case here. Thanks.